rs/bitcoin/ckbtc/minter/src/updates/update_balance.rs

```diff
@@ -283,60 +283,45 @@ pub async fn update_balance(
     Ok(utxo_statuses)
 }
 
-async fn kyt_check_utxo(
-    caller: Principal,
-    utxo: &Utxo,
-) -> Result<(String, UtxoCheckStatus, Principal), UpdateBalanceError> {
-    let kyt_principal = read_state(|s| {
-        s.kyt_principal
-            .expect("BUG: upgrade procedure must ensure that the KYT principal is set")
+async fn kyt_check_utxo(utxo: &Utxo) -> Result<UtxoCheckStatus, UpdateBalanceError> {
+    let new_kyt_principal = read_state(|s| {
+        s.new_kyt_principal
+            .expect("BUG: upgrade procedure must ensure that the new KYT principal is set")
             .get()
             .into()
     });
 
-    if let Some((uuid, status, api_key_owner)) = read_state(|s| s.checked_utxos.get(utxo).cloned())
-    {
-        return Ok((uuid, status, api_key_owner));
+    if let Some((_, status, _)) = read_state(|s| s.checked_utxos.get(utxo).cloned()) {
```

Comment:
> Actually it starts to feel like this cache is perhaps no longer needed since the KYT canister also keeps a cache. Can we remove it (in another ticket)?

Yes, I think it makes sense. In particular the KYT canister should decide whether its cache is up-to-date and will automatically loose it upon upgrades.